### ISSUE TITLE
Scripting: Actions can't have checkboxes

### ISSUE THREAD

**Issue by eishiya** [mapeditor/tiled#3464]
If an Action is `checkable`, its icon is displayed instead of the checkbox, with a button-like look to indicate whether the Action is `checked`:
Default icon:
![image](https://user-images.githubusercontent.com/70545646/187041103-dfe554c3-e45f-4910-aa2c-d3e9a5507e04.png)
Custom icon:
![image](https://user-images.githubusercontent.com/70545646/187041180-e2ed7327-516c-4acd-9ec2-a6f4f3ae045e.png)

When an action with an icon is unchecked, it's hard to see that it even *can* be checked. The default checkbox does a much better job of communicating that an action is checkable.

This only affects scripted actions, as none of the native checkable actions have icons.
Since script authors can simply choose to not assign an icon, it's probably fine that those with custom icons are displayed this way, allowing authors to essentially implement custom toggle styles if they wish. However, there's no way to *not* have an icon since iconless actions are given the extension icon by default! As such, there's no way to display a regular checkbox.

Possible solutions:
- Do not show the default extension icon on Actions that are `checkable`
- Make the default extension icon small and have it overlap the corner of the checkbox (and possibly any assigned icon as well).

**Comment by saurabh-zz007** [comment]
I was trying to reproduce the bug locally on windows by writing a quick test script which has a checkbox.  But in my pc it shows the icon as well as the checkbox, no overlapping. If the overlapping occures in other systems I can remove that icon whenever the action is clickable, Does this seems a good solution?

<img width="338" height="350" alt="Image" src="https://github.com/user-attachments/assets/37ab1003-0831-48ad-9801-3539311e98fc" />

I think the ScriptedAction inside scriptedaction.cpp, the icon is forced in the action immediately on creation. Due to this the qts native checkbox is not rendering properly in some devices. 

I am testing the fix that connects to changed signal, which tells the ScriptedAction not to force icon if the action is clickable.

Could you assign this to me? I should have a PR shortly for review

**Comment by bjorn** [comment]
> I was trying to reproduce the bug locally on windows by writing a quick test script which has a checkbox. But in my pc it shows the icon as well as the checkbox, no overlapping.

Did you switch the style to "Native"? The screenshots by @eishiya were probably using "Tiled Fusion" style.

Of course rather than avoiding the puzzle piece icon for checkable actions, we could also consider modifying the style such that it will show both the checkbox and the icon, like in your screenshot.

**Comment by saurabh-zz007** [comment]
Yes you are absolutely right, I was switched to native theme for some previous issue check. Now I am getting the things on that screenshot in my system. I will revert the changes in scriptedactions.cpp, and work for it now. I will update the PR and tell you what I am doing for this shortly.

**Comment by saurabh-zz007** [comment]
Based on the PR review feedback, I completely pivoted the approach. Instead of modifying how the scripts generate actions, I went into TiledProxyStyle in tiledproxystyle.cpp and updated the UI engine itself.

The tiled fusion theme now properly renders both the native checkbox and the icon side by side with correct spacing. And I have updated my previous PR. Here is a quick review on both tiled fusion theme and Native theme.

<img width="368" height="357" alt="Image" src="https://github.com/user-attachments/assets/fdb9beff-aedb-42dc-a3ad-0924ee13aa67" />

<img width="327" height="353" alt="Image" src="https://github.com/user-attachments/assets/4054408e-8395-4d72-8b1d-3714d8680e7d" />

**Comment by saurabh-zz007** [comment]
Is there anything else I need to do from my side to move this forward?

**Comment by bjorn** [comment]
> Is there anything else I need to do from my side to move this forward?

It might take a bit until I will get around to reviewing your changes. In the meantime though, from the screenshots I've noticed the inclusion of both the checkbox and the icon leads to a misalignment of the text, which has been solved in the Native theme by shifting the text and icon for all menu items when there is a checkbox. I realize this might be a little tricky, but maybe you'd like to look into how this is achieved?

**Comment by saurabh-zz007** [comment]
Sorry for being late here. 

> > Is there anything else I need to do from my side to move this forward?
> 
> It might take a bit until I will get around to reviewing your changes. In the meantime though, from the screenshots I've noticed the inclusion of both the checkbox and the icon leads to a misalignment of the text, which has been solved in the Native theme by shifting the text and icon for all menu items when there is a checkbox. I realize this might be a little tricky, but maybe you'd like to look into how this is achieved?

Yes I should consider that problem too. I did love to fix that thing.

**Comment by saurabh-zz007** [comment]
I have removed all the conditional statements which update the position of name and icon of only checkable options, now all the icons and their respective names get same position update as the checkable icon and name gets. Here are the results

<img width="365" height="369" alt="Image" src="https://github.com/user-attachments/assets/1c9c0f61-4f98-49b1-8a65-de9c20a3b47e" />

I have changed .gitignore file for something but undo that change is my most latest PR, so there is no change in there but still it is showing in file changes. 
Are the results satisfying?

**Comment by saurabh-zz007** [comment]
Does there is anything I missed?
